src/lib.rs

```diff
@@ -277,3 +259,57 @@ impl State {
         }
     }
 }
+
+fn join_users_to_room(
+    first_user: &User<Synching>,
+    second_user: &User<Synching>,
+    progress_bar: &ProgressBar,
+) -> tokio::task::JoinHandle<()> {
+    tokio::spawn({
+        let mut first_user = first_user.clone();
+        let mut second_user = second_user.clone();
+        let progress_bar = progress_bar.clone();
+
+        async move {
+            let room_created = first_user.create_room().await;
+            if let Some(room_id) = room_created {
+                first_user.join_room(&room_id).await;
+                second_user.join_room(&room_id).await;
+            } else {
+                log::info!("User {} couldn't create a room", first_user.id());
```

Comment:
I think this should also either `panic!` or be counted as part of the errors in the metrics. If OoS for this PR, let's leave a `TODO` here.

Reply:
Added TODO